Integrations/GoogleCloudStorage/GoogleCloudStorage.py

```diff
@@ -0,0 +1,369 @@
+#     if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) else demisto.params()['url']
+# # Should we use SSL
+# USE_SSL = not demisto.params().get('insecure', False)
+# # How many time before the first fetch to retrieve incidents
+# FETCH_TIME = demisto.params().get('fetch_time', '3 days')
+# # Service base URL
+# BASE_URL = SERVER + '/api/v2.0/'
+# # Headers to be sent in requests
+# HEADERS = {
+#     'Authorization': 'Token ' + TOKEN + ':' + USERNAME + PASSWORD,
+#     'Content-Type': 'application/json',
+#     'Accept': 'application/json'
+# }
+
+
+def safe_del(dictionary, key):
+    if key in dictionary:
+        del dictionary[key]
+
+
+# Remove proxy if not set to true in params
+if not USE_PROXY:
+    safe_del(os.environ, "HTTP_PROXY")
+    safe_del(os.environ, "HTTPS_PROXY")
+    safe_del(os.environ, "http_proxy")
+    safe_del(os.environ, "https_proxy")
+
+
+''' HELPER FUNCTIONS '''
+
+
+def init_storage_client():
+    cur_directory_path = os.getcwd()
+    credentials_file_name = "{}.json".format(demisto.uniqueFile())
+    credentials_file_path = os.path.join(cur_directory_path, credentials_file_name)
+    with open(credentials_file_path, "w") as creds_file:
+        json_object = json.loads(SERVICE_ACCOUNT_JSON)
+        json.dump(json_object, creds_file)
+    return storage.Client.from_service_account_json(credentials_file_path)
+
+
+def ec_key(path, *merge_by):
+
+    if len(merge_by) == 0:
+        return path
+
+    js_condition = ""
+    for key in merge_by:
+        if js_condition:
+            js_condition += " && "
+        js_condition += "val.{0} && val.{0} === obj.{0}".format(key)
+    return "{}({})".format(path, js_condition)
+
+
+def bucket2dict(bucket):
+    return {
+        "Name": bucket.name,
+        "TimeCreated": reformat_datetime_str(bucket._properties.get("timeCreated", "")),
+        "TimeUpdated": reformat_datetime_str(bucket._properties.get("updated", "")),
+        "OwnerID": "" if not bucket.owner else bucket.owner.get("entityId", "")
+    }
+
+
+def blob2dict(blob):
+    return {
+        "Name": blob.name,
+        "Bucket": blob.bucket.name,
+        "ContentType": blob.content_type,
+        "TimeCreated": datetime2str(blob.time_created),
+        "TimeUpdated": datetime2str(blob.updated),
+        "TimeDeleted": datetime2str(blob.time_deleted),
+        "Size": blob.size,
+        "MD5": blob.md5_hash,
+        "OwnerID": "" if not blob.owner else blob.owner.get("entityId", ""),
+        "CRC32c": blob.crc32c,
+        "EncryptionAlgorithm": blob._properties.get("customerEncryption", {}).get("encryptionAlgorithm", ""),
+        "EncryptionKeySHA256": blob._properties.get("customerEncryption", {}).get("keySha256", ""),
+    }
+
+
+def reformat_datetime_str(dt_str):
+    dt = None if not dt_str else datetime.datetime.strptime(dt_str, RFC3339_DATETIME_FORMAT)
+    return datetime2str(dt)
+
+
+def datetime2str(dt):
+    return "" if not dt else dt.strftime(DEMISTO_DATETIME_FORMAT)
```

Comment:
Can we have `0` as the input? If so we'll translate 0 -> "" which is incorrect...